### DISCUSSION TITLE
modifying PS1 causes some jank

### DISCUSSION THREAD

**Discussion by goolord** [zimfw/zimfw#498] [Q&A] [answered]
this plugin https://github.com/chisui/zsh-nix-shell stopped working after some recent updates, so I tried to roll my own solution

adding this line `PS1=\$($ZIM_HOME/modules/any-nix-shell/bin/nix-shell-info)$PS1`
to my zshrc results in some weird behavior when in a nix shell

https://user-images.githubusercontent.com/24906808/219551022-74dfc803-d04a-4b9e-a140-34fb3af50bc8.mov

https://github.com/goolord/dotfiles/tree/main/home/.config/zsh

does anyone have any ideas as to what's going wrong here?

**Comment by goolord** [comment, marked as answer]
seems to have something to do with the color of the output?

**Reply by goolord** [reply]
yes, that was the answer. i guess best practice is to use whatever this DSL is `%F{yellow}`, which is frustrating because I have no idea where this is documented anywhere, where it's specified, etc

**Reply by goolord** [reply]
i made something that works for my needs https://github.com/goolord/simple-zsh-nix-shell/blob/main/simple-zsh-nix-shell.zsh

**Reply by lnicola** [reply]
> i guess best practice is to use whatever this DSL is %F{yellow}, which is frustrating because I have no idea where this is documented anywhere, where it's specified, etc

Maybe [here](https://linux.die.net/man/1/zshmisc)?

**Reply by goolord** [reply]
nice, this will be useful to me. it wasn't obvious to me where to find this. maybe user error. lol
